features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java

```diff
@@ -136,16 +141,65 @@ public void testKafkaCollection() throws IOException {
         OnmsNode node = databasePopulator.getNode5();
         CollectionAgent agent = new MockCollectionAgent(node.getId(), "test", InetAddress.getLocalHost());
         NodeLevelResource nodeResource = new NodeLevelResource(node.getId());
+
         CollectionSet collectionSet = new CollectionSetBuilder(agent).withTimestamp(new Date(2))
                 .withNumericAttribute(nodeResource, "group1", "node5", 105, AttributeType.GAUGE)
-                .withNumericAttribute(nodeResource, "group2", "node5", 1050, AttributeType.GAUGE).build();
+                .withNumericAttribute(nodeResource, "group2", "node5", 1050, AttributeType.GAUGE)
+                .withNumericAttribute(new ResponseTimeResource(), "ICMP", "ICMP", 120, AttributeType.GAUGE).build();
         persister.visitCollectionSet(collectionSet);
         
-        await().atMost(1, TimeUnit.MINUTES).pollInterval(15, TimeUnit.SECONDS).until(() -> kafkaConsumer.getCollectionSet(), not(nullValue()));
-        assertThat(kafkaConsumer.getCollectionSet().getResource(0).getNode().getNodeId(), equalTo(node.getId().longValue()));
-        assertThat(kafkaConsumer.getCollectionSet().getResource(0).getNumericCount(), equalTo(2));
-        assertThat(kafkaConsumer.getCollectionSet().getResource(0).getNumeric(1).getValue(), equalTo(1050.0));
+        await().atMost(1, TimeUnit.MINUTES).pollInterval(15, TimeUnit.SECONDS).until(() -> kafkaConsumer.getCollectionSetValues(), not(Matchers.empty()));
+        assertThat(kafkaConsumer.getCollectionSetValues().get(0).getResource(0).getNode().getNodeId(), equalTo(node.getId().longValue()));
+        assertThat(kafkaConsumer.getCollectionSetValues().get(0).getResource(0).getNumericCount(), equalTo(2));
+        assertThat(kafkaConsumer.getCollectionSetValues().get(0).getResource(1).getResponse().getInstance(), equalTo(IP_ADDRESS));
+
+    }
+
+    private class ResponseTimeResource implements Resource {
```

Comment:
Can we move this to the `org.opennms.netmgt.collection.support.builder` package? Would be useful there so we could reuse it.

Reply:
I ended up not using the LatencyTypeResource in the test but still kept it for future use.